src/actions.ts

```diff
@@ -0,0 +1,39 @@
+import { ADD_MESSAGE, CLEAR_MESSAGE, CLEAR_ALL_MESSAGES } from './constants';
+import { MessageInterface } from './factory';
+
+export interface AddMessageInterface {
+  type: string;
```

Comment:
If you set the type here to be same as `ADD_MESSAGE`.

You can omit the `as AddMessageInterface` in the reducer, as Typescript will be able to infer it because it's a discriminated union.

https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions

Reply:
Thanks for early feedback
I am loving ts now,

Reply:
Great stuff, glad you like it!